saliency/core/base.py

```diff
@@ -110,18 +110,22 @@ def GetSmoothedMask(self,
       magnitude: If true, computes the sum of squares of gradients instead of
                  just the sum. Defaults to true.
     """
-    stdev = stdev_spread * (np.max(x_value) - np.min(x_value))
 
+    stdev = stdev_spread * (np.max(x_value) - np.min(x_value))
     total_gradients = np.zeros_like(x_value, dtype=np.float32)
-    for _ in range(nsamples):
-      noise = np.random.normal(0, stdev, x_value.shape)
-      x_plus_noise = x_value + noise
-      grad = self.GetMask(x_plus_noise, call_model_function, call_model_args,
+    shape = (nsamples,) + x_value.shape
+    noisy_samples = np.zeros(shape)
+    for i in range(nsamples):
```

Comment:
You could get rid of the loop completely if you used np.repeat(...) and generating the noise mask for the batched shape.